pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,990 @@
+
+**Selecting multiple rows or columns**
+
+If you want to see the ages of the first 3 patients tested, change the `0` to the list `[0,1,2]`. This indicates that you want to see the data in rows indexed 0,1, and 2. Similarly you can change the single column to be a list of columns.
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing.loc[[0,1,2],[ "age", "gender"]])
+</script>
+</lia-keep>
+</div>
+
+**Viewing full rows and columns**
+
+If you want to show all of the data in a row or column, instead of a list you can use a colon `:` to indicate that you want to include everything.
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing.loc[:,["mrn","first_name","last_name"]])
+</script>
+</lia-keep>
+</div>
+<br>
+
+A colon before the comma will show you all rows of the columns you selected and a colon after the comma will return all columns of the given rows.
+
+**Your turn:** What do you think will happen if you put colons both before and after the comma? Change the code below to check if you were correct.
+
+<div class="python_data">
+<lia-keep>
+<script type="text/x-sage">
+print(covid_testing.loc[[2,3,4],:])
+</script>
+</lia-keep>
+</div>
+<br>
+
+<div class = "options">
+If you look at other people's code, you may see columns refered to with `data_frame["column_name"]` instead of `data_frame.loc[:,"column_name"]`. While this will frequently give the same output, i.e. show you the entire column, it can [raise errors if you use it to try to update data](https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html). For that reason we are focusing on the `.loc` method.
+</div>
+
+### Quiz: DataFrames
+
+@sage
+
+1. Complete the following code by replacing the two `__?__`s with your own code to find out the first and last name of the patient in row 11942.
+
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+import pandas as pd
+covid_testing = pd.__?__('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+
+print(covid_testing.__?__[11942, ["first_name","last_name"]])
+</script>
+</lia-keep>
+</div>
+<br>
+
+Enter their name below to check your answer:
+
+[[grazdan greyjoy]]
+<script>
+  let input = "@input".trim().toLowerCase();
+  input == "grazdan greyjoy";
+</script>
+***
+<div class = "answer">
+
+On line 2, the command `read_csv` will convert the csv file's data into a DataFrame.
+
+On line 4, the `.loc` method will request the data from row 11942 and columns `"first_name"` and `"last_name"`.
+
+The completed code should look like this:
+
+```
+import pandas as pd
+covid_testing = pd.read_csv('https://raw.githubusercontent.com/arcus/education_modules/embedded_code/a_sample_module_template/covid_testing.csv')
+
+print(covid_testing.loc[11942, ["first_name","last_name"]])
```

Comment:
```suggestion
print(covid_testing.loc[11942, ["first_name","last_name"]])

```

The final line is getting slightly cutoff by the answer box in my view. Adding a newline at the end will hopefully help.